### ISSUE TITLE
fail to build simpleitk 2.1.0 due to gdcm compilation issue with gcc 11

### ISSUE THREAD

**Issue by hubutui** [SimpleITK/SimpleITK#1435]
related log:
```text
In file included from /build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.cxx:14:
/build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h:67:32: error: no member named 'numeric_limits' in namespace 'std'
  return v < 0 ? 0 : (v > std::numeric_limits<T>::max() ? std::numeric_limits<T>::max() : v);
                          ~~~~~^
/build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h:67:47: error: 'T' does not refer to a value
  return v < 0 ? 0 : (v > std::numeric_limits<T>::max() ? std::numeric_limits<T>::max() : v);
                                              ^
/build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h:63:20: note: declared here
template <typename T>
                   ^
[ 70%] Building CXX object Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/CMakeFiles/gdcmMSFF.dir/gdcmUIDGenerator.cxx.o
/build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h:67:51: error: no member named 'max' in the global namespace
  return v < 0 ? 0 : (v > std::numeric_limits<T>::max() ? std::numeric_limits<T>::max() : v);
                                                ~~^
[ 70%] Building CXX object Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/CMakeFiles/gdcmMSFF.dir/gdcmUUIDGenerator.cxx.o
/build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h:67:64: error: no member named 'numeric_limits' in namespace 'std'
  return v < 0 ? 0 : (v > std::numeric_limits<T>::max() ? std::numeric_limits<T>::max() : v);
                                                          ~~~~~^
/build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h:67:79: error: 'T' does not refer to a value
  return v < 0 ? 0 : (v > std::numeric_limits<T>::max() ? std::numeric_limits<T>::max() : v);
                                                                              ^
/build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h:63:20: note: declared here
template <typename T>
                   ^
/build/python-simpleitk/src/python-simpleitk-2.1.0/_skbuild/linux-x86_64-3.9/cmake-build/sitk-sb/ITK/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h:67:83: error: no member named 'max' in the global namespace
  return v < 0 ? 0 : (v > std::numeric_limits<T>::max() ? std::numeric_limits<T>::max() : v);
```
this gdcm issue is fixed by upstream patch https://github.com/malaterre/GDCM/commit/4404b770be337bd0d5d3c2289abfd34426433db2. Upgrade gdcm src to this commit would fix it.

**Comment by dave3d** [comment]
I think you need to report it to the ITK repo to get them to update to a more recent GDCM.

**Comment by blowekamp** [comment]
Support for gcc 11 was added to GDCM in ITK with this PR: https://github.com/InsightSoftwareConsortium/ITK/pull/2618

If you set the CMake variable: `ITK_GIT_TAG` in a SimpleITK superbuild to a hash after this merge you can easily use a newer version of ITK with the SimpleITK Superbuild.

**Comment by hubutui** [comment]
thanks @blowekamp

**Comment by blowekamp** [comment]
I made a request for the GDCM patch to be included in ITK 5.2.1:
 https://discourse.itk.org/t/patches-for-itk-5-2-1/4323/3

**Comment by dave3d** [comment]
Just checked, and the patch is now in ITK 5.2.1.
